### ISSUE TITLE
Changing task for time spent works properly only once

### ISSUE THREAD

**Issue by jsabak** [Dolibarr/dolibarr#19860]
# Bug
When I try to change a task for time spent entry it  works only once. If I try to change task for another entry it (aparently) does nothing. When I reload the page or go to another page and come back, I can change the task for next entry, but again only once. 

### My diagnosis is that: 
Line 235:
   		`if (GETPOST('taskid', 'int') != $id) {		// GETPOST('taskid') is id of new task`
uses variable `$id` in comparison and overwrites it in the next line 
`$id = GETPOST('taskid', 'int');`
the after POST saving changed task `(GETPOST('taskid', 'int')` gives `$id` that has been saved and not ID of the task which time is edited currently. 

### My solution:
I have renamed `$id` variable in the line 236 to `$id_temp`:
`$id_temp  = GETPOST('taskid', 'int');`

and simlarly in the line 242:
`$object->fetch($id_temp, $ref);`

This solution seems to work. But I have not tested it thorougly. 

## Environment
- **Version**: 14.0.4
- **OS**: Linux dolibarr 5.4.140-1-pve #1 SMP PVE 5.4.140-1 (Wed, 08 Sep 2021 16:21:59 +0200) x86_64
- **Web server**: Apache/2.4.41 (Ubuntu)
- **PHP**: 7.4.3
- **Database**: PostgreSQL 12.9 (Ubuntu 12.9-0ubuntu0.20.04.1)
- **URL(s)**: /dolibarr/projet/tasks/time.php

## Expected and actual behavior
See below.

## Steps to reproduce the behavior

1. Enter "Time spent" tab for a project
2. Click "Modify" icon for an item in "List of time consumed on tasks of project"
3. Change task for the item
4. Click "Save button" 
5. Task disappears from the list
5. Click "Modify" icon for another item it the list
6. Change task for the item
7. Click "Save button"
8. Task is not removed from the list (but should be removed)

![image](https://user-images.githubusercontent.com/566741/149765128-3f59fbd0-ca14-409b-ae0b-a75fc030b3b4.png)

**Comment by isamuse2** [comment]
@jsabak 
I have also encountred this bug. thank you for the fix.
